apps/api/src/app/workflows-v2/usecases/sync-to-environment/sync-to-environment.usecase.ts

```diff
@@ -176,11 +176,8 @@ export class SyncToEnvironmentUseCase {
     }
   }
 
-  private findDatabaseIdInProdByExternalId(
-    existingWorkflowSteps: StepResponseDto[] | undefined,
-    step: StepResponseDto
-  ) {
-    return existingWorkflowSteps?.find((existingStep) => existingStep.stepId === step.stepId)?._id ?? step._id;
```

Comment:
Removed the fallback fixed the sync issue. The problem was that we used the origin `_id` as a fallback, which wasn't available in the target environment during upsert. This prevented us from locating the control entry to update.

All other changes are just renames to make the code easier to follow.

Reply:
Awesome! Do we maybe have a small e2e for syncing the changes between envs?

Reply:
we have some, let me check what is the test scope there.